### ISSUE TITLE
Improve CPU Resource Split Semantics

### ISSUE THREAD

**Issue by GuillaumeMercier** [hpc/quo-vadis#23]
The test program `test-mpi-scopes` seems to fail on my system.
I'm lauching this test with 4 MPI processes on 2 cores featuring 2 HW threads each (i.e. 4 HWT in total).
Here is the output:
```
mercierg@argos:~/Developpement/Git/QuoVadis/qv-build/tests$ mpiexec --report-bindings --display-map --tag-output -host argos -np 4  --use-hwthread-cpus --oversubscribe --bind-to hwthread ./test-mpi-scopes 
 Data for JOB [13928,1] offset 0 Total slots allocated 1

 ========================   JOB MAP   ========================

 Data for node: argos   Num slots: 1    Max slots: 0    Num procs: 4
        Process OMPI jobid: [13928,1] App: 0 Process rank: 0 Bound: socket 0[core 0[hwt 0]]:[B./..]
        Process OMPI jobid: [13928,1] App: 0 Process rank: 1 Bound: socket 0[core 0[hwt 1]]:[.B/..]
        Process OMPI jobid: [13928,1] App: 0 Process rank: 2 Bound: socket 0[core 1[hwt 0]]:[../B.]
        Process OMPI jobid: [13928,1] App: 0 Process rank: 3 Bound: socket 0[core 1[hwt 1]]:[../.B]

 =============================================================
[1,0]<stderr>:[argos:20261] MCW rank 0 bound to socket 0[core 0[hwt 0]]: [B./..]
[1,1]<stderr>:[argos:20261] MCW rank 1 bound to socket 0[core 0[hwt 1]]: [.B/..]
[1,2]<stderr>:[argos:20261] MCW rank 2 bound to socket 0[core 1[hwt 0]]: [../B.]
[1,3]<stderr>:[argos:20261] MCW rank 3 bound to socket 0[core 1[hwt 1]]: [../.B]
[1,1]<stdout>:[1] self_scope taskid is 0
[1,1]<stdout>:[1] self_scope ntasks is 1
[1,2]<stdout>:[2] self_scope taskid is 0
[1,2]<stdout>:[2] self_scope ntasks is 1
[1,3]<stdout>:[3] self_scope taskid is 0
[1,3]<stdout>:[3] self_scope ntasks is 1
[1,0]<stdout>:[0] self_scope taskid is 0
[1,0]<stdout>:[0] self_scope ntasks is 1
[1,3]<stdout>:[3] base_scope taskid is 3
[1,3]<stdout>:[3] base_scope ntasks is 4
[1,1]<stdout>:[1] base_scope taskid is 1
[1,1]<stdout>:[1] base_scope ntasks is 4
[1,2]<stdout>:[2] base_scope taskid is 2
[1,2]<stdout>:[2] base_scope ntasks is 4
[1,0]<stdout>:[0] base_scope taskid is 0
[1,0]<stdout>:[0] base_scope ntasks is 4
[1,2]<stdout>:[2] Number of NUMA in base_scope is 1
[1,0]<stdout>:[0] Number of NUMA in base_scope is 1
[1,3]<stdout>:[3] Number of NUMA in base_scope is 1
[1,1]<stdout>:[1] Number of NUMA in base_scope is 1
[1,2]<stderr>:
[1,2]<stderr>:main@209: qv_scope_split() failed (rc=Split error)
[1,0]<stderr>:
[1,0]<stderr>:main@209: qv_scope_split() failed (rc=Split error)
[1,1]<stderr>:
[1,1]<stderr>:main@209: qv_scope_split() failed (rc=Split error)
[1,3]<stderr>:
[1,3]<stderr>:main@209: qv_scope_split() failed (rc=Split error)
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[13928,1],0]
  Exit code:    1
--------------------------------------------------------------------------

```

**Comment by samuelkgutierrez** [comment]
Hi, @GuillaumeMercier. This is a known issue in the current split semantics that needs addressing. We currently enforce a splitting of CPU resources that is overly strict: each task must have a piece of the parent resource after a split. For example, if you have 4 tasks that are attempting to split 2 indivisible CPU resources into 4 pieces, a split will currently fail. Better semantics would split the 2 resources somehow and perhaps leave some tasks without a piece. If you run with fewer MPI processes (say `-np 2`) the test should run to completion. When I have time, I'll update the code to implement the nicer split semantics. This is a good issue to track progress on the larger issue.

**Comment by samuelkgutierrez** [comment]
Looks like this issue was fixed in or around #118.
